tools/gcs-bucket-mover/gcs_bucket_mover/configuration.py

```diff
@@ -0,0 +1,151 @@
+                ' script?')
+
+        self._source_project_credentials = service_account.Credentials.from_service_account_file(
+            self._source_service_account_json)
+        self._target_project_credentials = service_account.Credentials.from_service_account_file(
+            self._target_service_account_json)
+        print 'Using the following service accounts for GCS credentials: '
+        print('Source Project - {}'.format(
+            self._source_project_credentials.service_account_email))
+        print('Target Project - {}\n'.format(
+            self._target_project_credentials.service_account_email))
+
+        self._source_storage_client = storage.Client.from_service_account_json(
+            self._source_service_account_json)
+        self._target_storage_client = storage.Client.from_service_account_json(
+            self._target_service_account_json)
+        self._target_service_account_email = self.__get_service_account_email(
+            self._target_storage_client)
+
+        # Load the values passed in on the command line
+        self._bucket_name = conf.bucket_name
+        self._temp_bucket_name = self.__get_temp_bucket_name(
+            self._bucket_name, conf.tempBucketName)
+        self._source_project = conf.source_project
+        self._target_project = conf.target_project
+
+    @property
+    def source_project_credentials(self):
+        """Get the credentials object for the source project"""
+        return self._source_project_credentials
+
+    @property
+    def target_project_credentials(self):
+        """Get the credentials object for the target project"""
+        return self._target_project_credentials
+
+    @property
+    def source_storage_client(self):
+        """Get the storage client object for the source project"""
+        return self._source_storage_client
+
+    @property
+    def target_storage_client(self):
+        """Get the storage client object for the target project"""
+        return self._target_storage_client
+
+    @property
+    def target_service_account_email(self):
+        """Get the email account of the target project's service account"""
+        return self._target_service_account_email
+
+    @property
+    def bucket_name(self):
+        """Get the name of the bucket to be moved"""
+        return self._bucket_name
+
+    @property
+    def temp_bucket_name(self):
+        """Get the temporary bucket name to use in the target project"""
+        return self._temp_bucket_name
+
+    @property
+    def source_project(self):
+        """Get the name of the source project"""
+        return self._source_project
+
+    @property
+    def target_project(self):
+        """Get the name of the target project"""
+        return self._target_project
+
+    def __get_service_account_email(self, storage_client):  #Google guidelines differ from pylint. pylint: disable=no-self-use
```

Comment:
Instead of turning off the no-self-use check, this doesn't need to be a class method. It should be a local function instead.

Reply:
The latest google-cloud-stroage module release now has the PR with this method incorporated, so I switched to using it.